src/test/java/io/vertx/test/core/JsonObjectTest.java

```diff
@@ -1622,6 +1622,13 @@ public void testJsonObjectEquality2() {
   }
 
   @Test
+  public void testPutInstantAsObject() {
+    Object instant = Instant.now();
+    JsonObject jsonObject = new JsonObject();
+    jsonObject.put("instant", instant);
```

Comment:
You shouldn't since instants are converted to string according to the rfc, if you want it automatically casted then you must do getInstant(key)